### ISSUE TITLE
Code rendering for shell script breaks for dashes

### ISSUE THREAD

**Issue by relikd** [sbarex/QLMarkdown#71]
I don't know if this issue is related to QLMarkdown or a third party code syntax parser. But a text element in a path that contains a dash will be shown as if it were a command parameter. Even though there is no whitespace before.

![screen](https://user-images.githubusercontent.com/1321720/133908935-c2703252-02f9-489a-bd77-5f61e2a8e2e9.png)

**Comment by sbarex** [comment]
Yes, this is a problem with the third party highlighting library. Unfortunately shell scripts, due to their syntax, are very complex to process correctly.

**Comment by soreavis** [comment]
The mis-highlight comes from the Bash option rule in the bundled `langDefs/sh.lang`:

```lua
Regex=[[ \-\-?[\w\-]+ ]]
```

It matches a `-word` run regardless of what precedes it, so a hyphen inside a path or filename (`my-project`, `sub-path`, `file-name.txt`, `x86-64`, …) gets coloured as a command option.

Prepending `\B` (a not-a-word-boundary assertion) makes the dash open an option only when it actually starts a new word:

```lua
Regex=[[ \B\-\-?[\w\-]+ ]]
```

- `ls -la` — space before the dash is `\W\W` (no word boundary) → `\B` matches → still highlighted.
- `my-project` — `y-` is `\w\W` (a word boundary) → `\B` fails → no longer treated as an option.

You're right that the highlighting itself is the third-party library, but since QLMarkdown ships its own copy of the lang defs (`Resources/highlight/langDefs/sh.lang`, which already carries a couple of local tweaks), this can be fixed there without bumping the `highlight` submodule. `\B` is zero-width and Boost.Xpressive has no lookbehind, so it's a one-token change.

Opened #217 with the fix plus a before/after for the `vim /some/sub-path.folder/dir` example from the report — genuine short/long/glued options stay highlighted, only the in-word hyphens go back to plain text.
